hushline/static/js/vision.js

```diff
@@ -0,0 +1,63 @@
+document.addEventListener('DOMContentLoaded', function () {
+    document.getElementById('fileInput').addEventListener('change', handleFilesSelect, false);
+});
+
+async function handleFilesSelect(evt) {
+    const files = evt.target.files;
+    if (!files.length) return;
+
+    const resultsContainer = document.getElementById('results');
+    resultsContainer.innerHTML = ''; // Clear previous results
+
+    for (const file of files) {
+        await processFile(file, resultsContainer);
+    }
+}
+
+function escapeHTML(str) {
+    return str.replace(/[&<>"']/g, function (m) {
```

Comment:
It's a list of characters that needs to be escaped to prevent it being interpreted as HTML - it comes from the OWASP cheat sheet: https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#output-encoding-for-html-contexts

Reply:
Since we can load an image of code, in theory code could be executed, but in the example below it shows no execution.

Reply:
Yeah, that makes sense, but reading that link it prefaces the list with `Here are some examples of encoded values for specific characters:`, while recommending using the `.textContent` attribute to ensure safe encoding.

Reply:
Looking further, libraries like `lodash` seem to use the method you've implemented for escaping unsafe content, https://github.com/lodash/lodash/blob/6a2cc1dfcf7634fea70d1bc5bd22db453df67b42/src/escape.ts#L41. I think this is ok to keep as is.